python/cudf/cudf/core/reshape.py

```diff
@@ -1160,3 +1160,273 @@ def _length_check_params(obj, columns, name):
+                names.append(arr.name)
+            else:
+                names.append(f"{prefix}_{i}")
+    else:
+        if len(names) != len(arrs):
+            raise ValueError("arrays and names must have the same length")
+        if not isinstance(names, list):
+            names = list(names)
+
+    return names
+
+
+def crosstab(
+    index,
+    columns,
+    values=None,
+    rownames=None,
+    colnames=None,
+    aggfunc=None,
+    margins=False,
+    margins_name="All",
+    dropna=None,
+    normalize=False,
+):
+    """
+    Compute a simple cross tabulation of two (or more) factors. By default
+    computes a frequency table of the factors unless an array of values and an
+    aggregation function are passed.
+
+    Parameters
+    ----------
+    index : array-like, Series, or list of arrays/Series
+        Values to group by in the rows.
+    columns : array-like, Series, or list of arrays/Series
+        Values to group by in the columns.
+    values : array-like, optional
+        Array of values to aggregate according to the factors.
+        Requires `aggfunc` be specified.
+    rownames : list of str, default None
+        If passed, must match number of row arrays passed.
+    colnames : list of str, default None
+        If passed, must match number of column arrays passed.
+    aggfunc : function, optional
+        If specified, requires `values` be specified as well.
+    margins : Not supported
+    margins_name : Not supported
+    dropna : Not supported
+    normalize : Not supported
+
+    Returns
+    -------
+    DataFrame
+        Cross tabulation of the data.
+
+    Examples
+    --------
+    >>> a = cudf.Series(["foo", "foo", "foo", "foo", "bar", "bar",
+    ...               "bar", "bar", "foo", "foo", "foo"], dtype=object)
+    >>> b = cudf.Series(["one", "one", "one", "two", "one", "one",
+    ...               "one", "two", "two", "two", "one"], dtype=object)
+    >>> c = cudf.Series(["dull", "dull", "shiny", "dull", "dull", "shiny",
+    ...               "shiny", "dull", "shiny", "shiny", "shiny"],
+    ...              dtype=object)
+    >>> cudf.crosstab(a, [b, c], rownames=['a'], colnames=['b', 'c'])
+    b   one        two
+    c   dull shiny dull shiny
+    a
+    bar    1     2    1     0
+    foo    2     2    1     2
+    """
+    if normalize is not False:
+        raise NotImplementedError("normalize is not supported yet")
+
+    if values is None and aggfunc is not None:
+        raise ValueError("aggfunc cannot be used without values.")
+
+    if values is not None and aggfunc is None:
+        raise ValueError("values cannot be used without an aggfunc.")
+
+    if not isinstance(index, (list, tuple)):
+        index = [index]
+    if not isinstance(columns, (list, tuple)):
+        columns = [columns]
+
+    rownames = (
+        _get_pivot_names(index, rownames, prefix="row")
+        if not rownames
+        else rownames
+    )
```

Comment:
Refactor: `x = y if cond else x` is better written as `if cond: x = y`.

Do the same for `colnames`.
```suggestion
    if not rownames:
        rownames = _get_pivot_names(index, rownames, prefix="row")
```

Reply:
Done.